### ISSUE TITLE
Define  ` ∩` and `∪` for `BareInterval`

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#650]
Since `BareInterval` is not a subtype of `Real`, I think it is OK to overload `Base.∩` and `Base.∪` for `BareInterval`s. 
Is there a reason I'm missing why we shouldn't do this?

I need this for `IntervalConstraintProgramming.jl`; I can also define them there if preferred.

**Comment by OlivierHnt** [comment]
We removed those also because of silence errors. `Base.∪` and `Base.∩` can be found in generic code for both scalars (not just numbers) and containers, so re-defining it with the interval definition is conflicting.
It is unfortunate how prevalent the `union`, `intersect`, etc, are in Base.

May I ask why you said that you need it? What is wrong with `intersect_interval` and `hull`?

**Comment by dpsanders** [comment]
Hmm I see.

I need them so I can write readable code :)

**Comment by OlivierHnt** [comment]
hehe yes I can sympathise with that :)

In this circonstance, I think the best is to add the following, *without importing* `∩` and `∪` from Base, 
```Julia
∩(a::BareInterval, b::BareInterval) = intersect_interval(a, b)
∪(a::BareInterval, b::BareInterval) = hull(a, b)
```
in [IntervalConstraintProgramming.jl](https://github.com/JuliaIntervals/IntervalConstraintProgramming.jl)
Then, you will have your own `∩` and `∪` inside the module, without conflicting with Base.
Of course outside the scope of the module this may still be confusing to have both `IntervalConstraintProgramming.jl.:∩` and `Base.:∩`.

**Comment by dpsanders** [comment]
Huh, interesting idea, thanks!

**Comment by dpsanders** [comment]
Another possibility is to use `\sqcup` and `\sqcap` (square versions of the intersection and union symbols), which have no meaning in Julia but are available as symbols that can be defined. Indeed `\sqcap` is one of the standard symbols for "interval hull"!

**Comment by OlivierHnt** [comment]
Yes we can add symbols to `IntervalArithmetic.Symbols`. `⊔`, `⊓` or  even `⊍` (`\cupdot`), ⩀ (`\capdot`).
